src/CMakeLists.txt

```diff
@@ -262,128 +334,186 @@ endif()
 # solvespace graphical executable
 
 if(ENABLE_GUI)
-    add_executable(solvespace WIN32 MACOSX_BUNDLE
-        ${solvespace_core_gl_SOURCES}
-        platform/entrygui.cpp
-        $<TARGET_PROPERTY:resources,EXTRA_SOURCES>)
-
-    add_dependencies(solvespace
-        resources)
-
-    target_link_libraries(solvespace
-        PRIVATE
-        solvespace-core
-        ${OPENGL_LIBRARIES})
-
-    # OpenGL version
-    if(OPENGL STREQUAL 3)
-        target_sources(solvespace PRIVATE
-            render/gl3shader.cpp
-            render/rendergl3.cpp)
-    elseif(OPENGL STREQUAL 1)
-        target_sources(solvespace PRIVATE
-            render/rendergl1.cpp)
-    else()
-        message(FATAL_ERROR "Unsupported OpenGL version ${OPENGL}")
-    endif()
-
-    # Platform-specific
-    if(WIN32)
-        target_sources(solvespace PRIVATE
-            platform/guiwin.cpp)
-
-        target_link_libraries(solvespace PRIVATE comctl32)
-    elseif(APPLE)
-        target_compile_options(solvespace PRIVATE -fobjc-arc)
-        target_compile_definitions(solvespace PRIVATE GL_SILENCE_DEPRECATION)
-
-        target_sources(solvespace PRIVATE
-            platform/guimac.mm)
-        set_target_properties(solvespace PROPERTIES
-            OUTPUT_NAME SolveSpace
-            XCODE_ATTRIBUTE_ENABLE_HARDENED_RUNTIME "YES"
-            XCODE_ATTRIBUTE_PRODUCT_BUNDLE_IDENTIFIER "com.solvespace"
-            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
-    elseif(EMSCRIPTEN)
-        set(SHELL ${CMAKE_CURRENT_SOURCE_DIR}/platform/html/emshell.html)
-        set(LINK_FLAGS
-            --bind --shell-file ${SHELL}
-            --no-heap-copy -s ALLOW_MEMORY_GROWTH=1 -s WASM=1 -s ASYNCIFY=1
-            -s DYNCALLS=1 -s ASSERTIONS=1
-            -s TOTAL_STACK=33554432 -s TOTAL_MEMORY=134217728)
-
-        get_target_property(resource_names resources NAMES)
-        foreach(resource ${resource_names})
-            list(APPEND LINK_FLAGS --preload-file ${resource})
-        endforeach()
-
-        if(CMAKE_BUILD_TYPE STREQUAL Debug)
-            list(APPEND LINK_FLAGS
-                --emrun --emit-symbol-map
-                -s DEMANGLE_SUPPORT=1
-                -s SAFE_HEAP=1)
-        endif()
-
-        target_sources(solvespace PRIVATE
-            platform/guihtml.cpp)
-
-        string(REPLACE ";" " " LINK_FLAGS "${LINK_FLAGS}")
-        set_target_properties(solvespace PROPERTIES
-            LINK_FLAGS "${LINK_FLAGS}")
-        set_source_files_properties(platform/guihtml.cpp PROPERTIES
-            OBJECT_DEPENDS ${SHELL})
+    if(NOT USE_QT_GUI)
```

Comment:
Definitely not like this. The QT conditional things should be integrated "fine grained" instead of a "big" if block and a bunch of copy-paste.
Take a look how `EMSCRIPTEN` was done - it was the lat platform added.